analyzers/src/SonarAnalyzer.VisualBasic/Facade/VisualBasicFacade.cs

```diff
@@ -51,9 +51,16 @@ internal sealed class VisualBasicFacade : ILanguageFacade<SyntaxKind>
         node.FindConstantValue(model);
 
     public IMethodParameterLookup MethodParameterLookup(SyntaxNode invocation, IMethodSymbol methodSymbol) =>
-        invocation?.ArgumentList() is { } argumentList
-            ? new VisualBasicMethodParameterLookup(argumentList, methodSymbol)
-            : null;
+        invocation switch
+        {
+            null => null,
+            AttributeSyntax x => new VisualBasicAttributeParameterLookup(x.ArgumentList.Arguments, methodSymbol),
+            IdentifierNameSyntax
+            {
+                Parent: NameColonEqualsSyntax { Parent: SimpleArgumentSyntax { IsNamed: true, Parent.Parent: AttributeSyntax attribute } }
+            } => new VisualBasicAttributeParameterLookup(attribute.ArgumentList.Arguments, methodSymbol),
+            _ => new VisualBasicMethodParameterLookup(invocation.ArgumentList(), methodSymbol),
```

Comment:
The `ArgumentList` can also return null. In that case, the `VisualBasicMethodParameterLookup` will throw.

Reply:
This was already under test before:
https://github.com/SonarSource/sonar-dotnet/blob/5970cedb1ca7c307a27a43e3e572d3e7fe429d80/analyzers/tests/SonarAnalyzer.Test/Facade/VisualBasicFacadeTest.cs#L104-L121

`ArgumentList()` throws:
https://github.com/SonarSource/sonar-dotnet/blob/5970cedb1ca7c307a27a43e3e572d3e7fe429d80/analyzers/src/SonarAnalyzer.VisualBasic/Extensions/SyntaxNodeExtensions.cs#L118

Reply:
.. and I said: It is already throwing in current master. I didn't changed that:
This is how the code looks like on master:

https://github.com/SonarSource/sonar-dotnet/blob/75e2d7b1c0cacf689ca42c8cc4d67f94a579ac43/analyzers/src/SonarAnalyzer.VisualBasic/Facade/VisualBasicFacade.cs#L59-L61

I can change it, but I would do so in a separate PR.

Reply:
We discussed offline, `invocation.ArgumentList()` returns null only if the invocation is null. This will not happen due to the previous check.